src/expr/src/scalar/func.rs

```diff
@@ -2039,6 +2039,36 @@ where
     }
 }
 
+fn date_trunc_interval<'a>(a: Datum, int: Interval) -> Result<Datum<'a>, EvalError>
+{
+    use mz_repr::adt::datetime::DateTimeField;
+    let units = a.unwrap_str();
+    units.parse()
+        .map_err(|_| EvalError::UnknownUnits(units.to_owned()))
+        .and_then(|dtu|
+            match dtu {
+                DateTimeUnits::Millennium => Ok(DateTimeField::Millennium),
+                DateTimeUnits::Century => Ok(DateTimeField::Century),
+                DateTimeUnits::Decade => Ok(DateTimeField::Decade),
+                DateTimeUnits::Year => Ok(DateTimeField::Year),
+                DateTimeUnits::Month => Ok(DateTimeField::Month),
+                DateTimeUnits::Day => Ok(DateTimeField::Day),
+                DateTimeUnits::Hour => Ok(DateTimeField::Hour),
+                DateTimeUnits::Minute => Ok(DateTimeField::Minute),
+                DateTimeUnits::Second => Ok(DateTimeField::Second),
+                DateTimeUnits::Milliseconds => Ok(DateTimeField::Milliseconds),
+                DateTimeUnits::Microseconds => Ok(DateTimeField::Microseconds),
+                other => Err(EvalError::Undefined(format!("truncate interval by \"{}\"", other)))
```

Comment:
`DateTimeField` has a `FromStr` implementation; could you use it to avoid parsing into `DateTimeUnits`?

Reply:
yes, good point. I do  that